pyart/graph/gridmapdisplay.py

```diff
@@ -328,11 +653,65 @@ def _check_basemap(self):
         if self.basemap is None:
             raise ValueError('no basemap plotted')
 
+    def _make_basemap(self, resolution='l', area_thresh=10000,
+                      auto_range=True, min_lon=-92, max_lon=-86,
+                      min_lat=40, max_lat=44, ax=None):
+        """
+        Make a basemap.
+
+        Parameters
+        ----------
+        auto_range : bool
+            True to determine map ranges from the grid_lats and grid_lons
+            attribute.  False will use the min_lon, max_lon, min_lat, and
+            max_lat parameters for the map range.
+        min_lat, max_lat, min_lon, max_lon : float
+            Latitude and longitude ranges for the map projection region in
+            degrees.  These parameter are not used if auto_range is True.
+        resolution : 'c', 'l', 'i', 'h', or 'f'.
+            Resolution of boundary database to use. See Basemap documentation
+            for details.
+        area_thresh : int
+            Basemap area_thresh parameter. See Basemap documentation.
+        ax : axes or None.
+            Axis to add the basemap to, if None the current axis is used.
+
+        """
+        # parse the parameters
+        ax = common.parse_ax(ax)
+
+        # determine map region
+        if auto_range:
+            max_lat = self.grid_lats.max()
+            max_lon = self.grid_lons.max()
+            min_lat = self.grid_lats.min()
+            min_lon = self.grid_lons.min()
+
+        if self.debug:
+            print("Maximum latitude: ", max_lat)
+            print("Maximum longitude: ", max_lon)
+            print("Minimum latitude: ", min_lat)
+            print("Minimum longitute: ", min_lon)
+
+        # determine plot center
+        lat_0 = self.grid.axes['lat']['data'][0]
+        lon_0 = self.grid.axes['lon']['data'][0]
+
+        # plot the basemap
+        basemap = Basemap(llcrnrlon=min_lon, llcrnrlat=min_lat,
+                          urcrnrlon=max_lon, urcrnrlat=max_lat,
+                          lat_0=lat_0, lon_0=lon_0,
+                          projection='tmerc', area_thresh=area_thresh,
```

Comment:
I prefer mercator to miller as most maps (e.g google maps) use it and it also perverse format. `'tmerc'` work better than `'merc'`, setting the scale from origin and it would also set the axes origin if it were not for [this basemap bug](https://github.com/matplotlib/basemap/issues/192)